.github/workflows/generate_doc.yml

```
@@ -44,48 +52,55 @@ jobs:
   deploy-doc:
     runs-on: ubuntu-latest
     needs: build-doc
-    if: startsWith(github.ref, 'refs/tags/')
```

```suggestion
    if: startsWith(github.ref, 'refs/tags/') ||  github.ref == 'refs/heads/main'
```

Missing condition to not deploy on PRs